assets/js/modules/adsense/components/common/AdBlockingRecoveryToggle.test.js

```diff
@@ -0,0 +1,226 @@
+import { enabledFeatures } from '../../../../features';
+
+describe( 'AdBlockingRecoveryToggle', () => {
+	const validSettings = {
+		accountID: 'pub-12345678',
+		clientID: 'ca-pub-12345678',
+		useSnippet: false,
+		accountStatus: ACCOUNT_STATUS_READY,
+		siteStatus: SITE_STATUS_READY,
+		adBlockingRecoverySetupStatus: '',
+	};
+
+	it.each( [
+		[ 'the Ad blocker detection feature flag is not enabled', false ],
+		[ 'Ad blocker recovery setup status is empty', true ],
+	] )(
+		'should not render the Ad blocking recovery toggle when %s',
+		( testName, adBlockerDetectionEnabled ) => {
+			const { container } = render( <AdBlockingRecoveryToggle />, {
+				setupRegistry: ( registry ) => {
+					if ( adBlockerDetectionEnabled ) {
+						enabledFeatures.add( 'adBlockerDetection' );
+					}
+					provideModules( registry, [
+						{
+							slug: 'adsense',
+							active: true,
+							connected: true,
+						},
+					] );
+					registry
+						.dispatch( MODULES_ADSENSE )
+						.receiveGetSettings( validSettings );
+				},
+			} );
+
+			expect(
+				container.querySelector(
+					'.googlesitekit-settings-module__ad-blocking-recovery-toggles'
+				)
+			).toBeNull();
+
+			expect( container ).toBeEmptyDOMElement();
+		}
+	);
+
+	it( 'should render the Ad blocking recovery toggle when the conditions are met', () => {
+		const { container, getByLabelText, getAllByRole } = render(
+			<AdBlockingRecoveryToggle />,
+			{
+				setupRegistry: ( registry ) => {
+					enabledFeatures.add( 'adBlockerDetection' );
+					provideModules( registry, [
+						{
+							slug: 'adsense',
+							active: true,
+							connected: true,
+						},
+					] );
+					registry.dispatch( MODULES_ADSENSE ).receiveGetSettings( {
+						...validSettings,
+						adBlockingRecoverySetupStatus:
+							AD_BLOCKING_RECOVERY_SETUP_STATUS_TAG_PLACED,
+						useAdBlockerDetectionSnippet: true,
+						useAdBlockerDetectionErrorSnippet: true,
+					} );
+				},
+			}
+		);
+
+		expect(
+			container.querySelector(
+				'.googlesitekit-settings-module__ad-blocking-recovery-toggles'
+			)
+		).not.toBeNull();
+
+		expect( container.textContent ).toContain( 'Ad blocking recovery' );
+		expect(
+			getByLabelText( /Place ad blocking recovery tag/i )
+		).toBeInTheDocument();
+
+		// The Material Design switch component is represented
+		// by multiple elements including a 'div' and an 'input'.
+		// We loop over all these to verify that they are checked.
+		const recoveryTagSwitchElements = getAllByRole( 'switch', {
+			name: /place ad blocking recovery tag/i,
+		} );
+		recoveryTagSwitchElements.forEach( ( switchEl ) => {
+			expect( switchEl ).toBeChecked();
+		} );
+		const errorProtectionTagSwitchElements = getAllByRole( 'switch', {
+			name: /place error protection tag/i,
+		} );
+		errorProtectionTagSwitchElements.forEach( ( switchEl ) => {
+			expect( switchEl ).toBeChecked();
+		} );
+	} );
+
+	it( 'should render the Ad blocking recovery tag toggle unchecked', () => {
```

Comment:
Can you update this test name to properly describe what's being tested here. ie. error protection toggle is not shown when ad blocking recovery toggle is unchecked.

Reply:
Updated.